enf/dns_zone_test.go

```diff
@@ -0,0 +1,205 @@
+				"enf_domain": "N/n0",
+				"id": "1234",
+				"zone_domain_name": "abc.def",
+				"modified": null
+			}
+		]
+	}
+	`
+
+	createdTime, _ := time.Parse(time.RFC3339, "2019-10-21T19:48:21.961747Z")
+
+	expected := &Zone{
+		Created:        Time(createdTime),
+		Description:    String("Test zone"),
+		EnfDomain:      String("N/n0"),
+		ID:             String("1234"),
+		Modified:       nil,
+		ZoneDomainName: String("abc.def"),
+	}
+
+	method := func(client *Client) (interface{}, *http.Response, error) {
+		return client.DNS.GetZone(context.Background(), "1234")
+	}
+
+	testParams := &TestParams{
+		Path:             path,
+		RequestBody:      struct{}{},
+		ResponseBodyMock: responseBodyMock,
+		Expected:         expected,
+		Method:           method,
+		T:                t,
+	}
+
+	getTest(testParams)
+}
+
+func TestDNSService_CreateZone(t *testing.T) {
+	path := "/api/xdns/2019-05-27/zones"
+
+	requestBody := &CreateZoneRequest{
+		Description:    String("Test create zone"),
+		ZoneDomainName: String("abc.def"),
+	}
+
+	responseBodyMock := `{
+		"data": [
+			{
+				"description": "Test create zone",
+				"enf_domain": "N/n0",
+				"id": "1234",
+				"zone_domain_name": "abc.def"
+			}
+		]
+	}
+	`
+
+	expected := &Zone{
+		Description:    String("Test create zone"),
+		EnfDomain:      String("N/n0"),
+		ID:             String("1234"),
+		ZoneDomainName: String("abc.def"),
+	}
+
+	method := func(client *Client) (interface{}, *http.Response, error) {
+		return client.DNS.CreateZone(context.Background(), requestBody)
+	}
+
+	testParams := &TestParams{
+		Path:             path,
+		RequestBody:      requestBody,
+		ResponseBodyMock: responseBodyMock,
+		Expected:         expected,
+		Method:           method,
+		T:                t,
+	}
+
+	postTest(testParams)
+}
+
+func TestDNSService_UpdateZone(t *testing.T) {
+	path := "/api/xdns/2019-05-27/zones/1234"
+
+	requestBody := &UpdateZoneRequest{
+		Description: String("Test update zone"),
+	}
+
+	responseBodyMock := `{
+		"data": [
+			{
+				"description": "Test update zone",
+				"enf_domain": "N/n0",
+				"id": "1234",
+				"zone_domain_name": "abc.def"
+			}
+		]
+	}`
+
+	expected := &Zone{
+		Description:    String("Test update zone"),
+		EnfDomain:      String("N/n0"),
+		ID:             String("1234"),
+		ZoneDomainName: String("abc.def"),
+	}
+
+	method := func(client *Client) (interface{}, *http.Response, error) {
+		return client.DNS.UpdateZone(context.Background(), "1234", requestBody)
+	}
+
+	testParams := &TestParams{
+		Path:             path,
+		RequestBody:      requestBody,
+		ResponseBodyMock: responseBodyMock,
+		Expected:         expected,
+		Method:           method,
+		T:                t,
+	}
+
+	putTest(testParams)
+}
+
+func TestDNSService_DeleteZone(t *testing.T) {
+	path := "/api/xdns/2019-05-27/zones/1234"
+
+	method := func(client *Client) (interface{}, *http.Response, error) {
+		resp, err := client.DNS.DeleteDNSZone(context.Background(), "1234")
```

Comment:
One last one!! :grin:

Reply:
Oh my haha. My eyes are playing tricks (and I'm rushing a bit). Thanks for the catch!